sdk/keyvault/azure-security-keyvault-keys/src/recover_deleted_key_operation.cpp

```diff
@@ -17,10 +18,14 @@ Azure::Security::KeyVault::Keys::RecoverDeletedKeyOperation::PollInternal(
   std::unique_ptr<Azure::Core::Http::RawResponse> rawResponse;
   if (!IsDone())
   {
-    rawResponse = m_pipeline->Send(
-        context,
-        Azure::Core::Http::HttpMethod::Get,
-        {_detail::KeysPath, m_value.Name(), m_value.Properties.Version});
+    try
+    {
+      rawResponse = m_keyClient->GetKey(m_value.Name(), {}, context).RawResponse;
+    }
+    catch (Azure::Core::RequestFailedException& error)
+    {
+      rawResponse = std::move(error.RawResponse);
+    }
```

Comment:
I am not sure if swallowing RFE is a good idea here (or anywhere for that matter).

Reply:
what is RFE?

Reply:
request failed exception

Reply:
I can explain it:

Operations "DeleteKey" and "RecoverKey" are not pure-LROs.  This means, the server won't provide a mechanism to return an operation model or a way to poll for updates like it happen with pure-LROs (like copyBlob).

In this case, the SDK client is creating the LRO directly.  For example, when a Key is deleted, the server will take some time for moving the Key to the /deletetedKeys path.  The way the LRO works in this case is by trying to get the key from the deletedKeys on each call to Poll().  If getting the Key returns a RequestException with code 404, it means that the key is not yet found in the deletedKeys. The server is still working on it. 

That's way, on Poll(), getting a request fail exception is fine because we are using the client to try to get the key from someplace where it will eventually appear.  

Before this change, the way I was avoiding the try/catch was by not using a key client to call Poll().  Instead, I was using the http pipeline from the client directly to send the request and get a rawResponse without using a key client operation which would throw when the key is not found.

The problem with doing such, combined with the new approach of the Operation ctor, is that now the Operation takes a keyClient to be created, and it can steal or use the internal pipeline from the client, so it needs to rely on the Client public operations to do the Polling().  :)

Hope this explanation helps.  Otherwise, talk to Rick, hahaha,, kidding, let me know and I can explain in a call.

Reply:
Can you add a concise version of this as a comment in source, especially since this seems to be KeyVault specific. We don't want folks in other client library SDKs to start adopting this pattern, unless it makes sense for that particular service.